test/unit/data.js

```diff
@@ -874,3 +874,18 @@ test( "Check that the expando is removed when there's no more data", function()
 		}
 	}
 });
+
+test( ".data(prop) does not create expando", function() {
+	expect( 2 );
+
+	var div = jQuery( "<div/>" );
+
+	div.data("foo");
+	equal( false, jQuery.hasData( div[0] ) );
+	// Make sure no expando has been added
+	for ( key in div[ 0 ] ) {
```

Comment:
```js
for ( var key in div[ 0 ] )
```

I really don't like this form. :) `var` is function scoped and we write it outside of any blocks inside a function to not suggest it behaves otherwise.

(i.e. this should be done exactly as this PR does it)